Sib 177

marwisni opened the pull request:
Added chart and tests.

JakubMakarewiczCap committed:
SIB-177 Bug fixes
- Fixed Mariusz's bugs on Mariusz's branch
- Using CoinMarketCap for coininfo
- Using CoingGecko for coinMarketData

Commit:
SIB-177 Prepare backend to presenting chart

* fixed bugs in Translator
* added sparkline to getTopCoins method
* added separate updater for sparkline one time per hour

JakubMakarewiczCap committed:
SIB-177 Applied spotless, added chart to controller

agnieszkapiela committed:
SIP-177 fixed test and added spotless

Commit:
SIB-177 fixed and added tests.

* fixed Kuba's test.
* fixed another tests which stopped working after added chart functionality
* added more tests
* turned off adding coins which are not available on CMC in initalizer to not block CoinGecko api. Updater will make this stuff later in chunks if it will be necesarry.
* added more keys for CMC api

Commit:
SIB-177 fixed and added tests.

* changed order in price list:
  from the OLDEST to the LATEST
* change fictive timestamps to real:
  CoinGecko is updating sparkline only 1 per day about 7am Warsaw Time, so whole day after update you have the same chart.

21JFX commented on src/main/java/com/capgemini/fs/coindashboard/apiCommunicator/coinGeckoCommunicator/CoinGeckoFacade.java:
```diff
@@ -186,7 +191,18 @@ public Optional<Result> getCoinInfo(List<String> coins) {
+        if (c > 19) {
+          if (c == 20) {
+            log.info("{} Waiting 60 seconds to not block CoinGecko", c+1);
+            sleep(60000l);
```
https://stackoverflow.com/questions/54394042/java-how-to-avoid-using-thread-sleep-in-a-loop

21JFX commented on src/main/java/com/capgemini/fs/coindashboard/apiCommunicator/coinGeckoCommunicator/CoinGeckoFacade.java:
```diff
@@ -186,7 +191,18 @@ public Optional<Result> getCoinInfo(List<String> coins) {
+          }
+          //if ((c % 10) == 0) {
+            log.info("{} Waiting 3 seconds to not block CoinGecko", c+1);
+            sleep(3000l);
```
same as above

21JFX reviewed (commented):
to fix

21JFX committed:
SIB-177
-quick bug fix for quality gate in CoinGeckoFacade
Should be replaced in a future by ExecutorService due to BusyWaiting problem in a loop

21JFX committed:
SIB-177 Fixed my bugs
-quick bug fix for quality gate in CoinGeckoFacade
Should be replaced in a future by ExecutorService due to BusyWaiting problem in a loop

21JFX committed:
SIB-177 [Final] Fixed my other bugs pointed by SonarCloud
-quick bug fix for quality gate in CoinGeckoFacade
Should be replaced in a future by ExecutorService due to BusyWaiting problem in a loop

Commit:
SIB-177 Adjusted timestamps for chart.

* Adjusted timestamp for chart according CoinGecko is updating sparkline twice per day: 7am and 7pm Warsaw Time.
* Allowed up to 5 calls per 250 coins for CoinInfoDetails to CoinGecko.

Commit:
SIB-177 Hotfix bug in tests.
